### DISCUSSION TITLE
Touchpad has high latency

### DISCUSSION THREAD

**Discussion by azizsayfiddin** [hyprwm/Hyprland#6527] [Q&A]
What program is responsible for handling events from the touchpad on hyprland/wayland?

**Comment by vaxerski** [comment]
uhh libinput?

**Reply by azizsayfiddin** [reply]
Do you have a laptop with a touchpad you could test? When I used the touchpad in DWM, the experience felt noticeably smoother than with Wayland compositors. I suspect the issue might be related to libinput, but I'm not entirely sure. I’d like to try fixing it, though I don’t have experience programming in C, so I'm unsure how long it might take.

**Comment by fufexan** [comment]
Are you sure it's not an illusion because of Vsync?

**Reply by azizsayfiddin** [reply]
> Are you sure it's not an illusion because of Vsync?

No, because the mouse and trackpoint work fine.

**Reply by Alumniminium** [reply]
disable vfr and/or make your gpu run at a higher min-frequency

**Reply by azizsayfiddin** [reply]
Disabling and enabling VFR doesn't change anything. I don't think it's the rendering problem. In X DE/WM touchpad feels much much better than on Hyprland. Also in Sway, KDE, and Gnome, it's pretty fine, but still worse than on X DE/WMs. How can I make the latency of the touchpad lower?

**Comment by azizsayfiddin** [comment]
The delay has been greatly reduced.

**Comment by azizsayfiddin** [comment]
How can it be reduced? Why touchpad latency on X11 dwm feels much much lower than any wayland compositors?
